File: LLD_Java/BookMyShow/Booking/Booking.java
package Java.BookMyShow.Booking;
import java.util.ArrayList;
import java.util.List;
import Java.BookMyShow.Theatre.*;
public class Booking {
    static int counter=0;
    int bookID;
    List<Seat> bookedSeats = new ArrayList<>();
    Show show; // it will have movie and timmings(start, end)
    Payment payment;

    public int getBookID(){
        return bookID;
    }
    public void setBookID(){
        this.bookID = counter;
        counter+=1;
    }
    public Show getShow(){
        return show;
    }
    public void setShow(Show show){
        this.show=show;
    }
    public List<Seat> getBookedSeats(){
        return bookedSeats;
    }
    public void setBookedSeats(List<Seat> bookedSeats){
        this.bookedSeats=bookedSeats;
    }
    public Payment getPayment(){
        return payment;
    }
    public void setPayment(Payment payment){
        this.payment=payment;
    }
}

File: LLD(Python)/bookMyShow/Model/Booking.py
class Booking:
    __bid=0
    def __init__(self):
        # for user it is
        # bookedseats
        # show -> movie with hall detail
        self.__show = None
        self.__bookedSeats = []
        self.__bookingID = self.__bid
        self.__bid += 1

        # booking will have payment
        self.__payment = None

    def setShow(self, show):
        self.__show = show

    def getShow(Self):
        return self.__show

    def getBookedSeats(self):
        return self.__bookedSeats
    
    def setBookedSeats(self, bookedSeat):
        self.__bookedSeats.append(bookedSeat)

    def getPayment(self):
        return self.__payment
    
    def setPayment(self, payment):
        self.__payment.append(payment)





